golem-worker-executor-base/src/worker/mod.rs

```diff
@@ -1330,6 +1333,29 @@ impl<Ctx: WorkerCtx> Worker<Ctx> {
     }
 }
 
+pub fn merge_worker_env_with_component_env(
+    worker_env: Option<Vec<(String, String)>>,
+    component_env: HashMap<String, String>,
+) -> Vec<(String, String)> {
+    let mut seen_keys = HashSet::new();
+    let mut result = Vec::new();
+
+    for (key, value) in component_env {
+        seen_keys.insert(key.clone());
+        result.push((key, value));
+    }
+
+    if let Some(worker_env) = worker_env {
+        for (key, value) in worker_env {
+            if !seen_keys.contains(&key) {
```

Comment:
per-worker env vars should always override the base (per-component) ones, so this `if` should be removed